### PR TITLE
Significant docs cleanup and refresh

### LINKED ISSUE #2553: Formalize Project Naming Conventions
In recent PRs to k/website, @sftim has raised concerns about our choice to use "Gateway API" instead of "The Gateway API". We have not formally documented that decision or had detailed discussion about that preference. This issue intends to be a place for discussion and eventually documentation of our decision here.

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR is pretty massive, here's an attempt at a changelog:

**Logo and Design**
* Logo files have been added
* Logo replaces title in header
* Logo added to main page
* Header background/overall primary color changes to Kubernetes primary color
* Updated main diagram on front page and added link to file that can be used to edit

**Mesh**
* A new "Mesh" section has been added, including Overview, GAMMA, and Service Facets Pages. (None of this content is new, it's just all more neatly contained in a Mesh-specific section).
* Once this merges, `gateway-api.sigs.k8s.io/mesh` will get you to this section
* Redirects from old pages have been added

**Layout**
* Tabs now show up on small displays (previously they'd hide below ~1200px width).
* GEPs have been moved from "reference" to a new "enhancements" tab
* Blog tab/navigation has been hidden, but individual pages remain for the sake of any old links out there
* Sidebar navigation is now fully expanded by default (not sure about this one, but it mostly feels like an improvement)

**Docs Consistency**
* A new documentation style guide has been added to extend the Kubernetes upstream one
* Project name has been clarified as "Gateway API", references to "the Gateway API" have been corrected
* "Experimental" and "Standard" channel callouts have been standardized
* Absolute links have been replaced with relative ones wherever possible

**Misc**
* Style of FAQs has been cleaned up throughout, now questions are included in navigation 

**Which issue(s) this PR fixes**:
Fixes #2553

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by robscott** [comment]
Recommend spending some time looking at the deploy preview when reviewing this one: https://deploy-preview-2691--kubernetes-sigs-gateway-api.netlify.app/. 

Adding a hold since this is a big PR and want to give people a chance to look through it.

/hold

**Comment by youngnick** [comment]
Nice, this LGTM (as long as it merges after #2689 and _I_ don't have to rebase 😄 )

File: site-src/contributing/index.md
Comment (youngnick) [inline site-src/contributing/index.md:25]: Not sure how this happened, but this change doesn't work properly.

Comment (robscott) [reply site-src/contributing/index.md:25]: Good catch, thanks! I did a global find and replace for absolute links to transform them into relative ones but clearly messed up some places in the process, should be good to go now.

File: mkdocs.yml
Comment (youngnick) [inline mkdocs.yml:110]: Does it make more sense to have this section sorted in roughly _declining_ order of stability? I'm thinking:

1. Memorandum
2. Standard
3. Experimental
4. Implementable
5. Provisional

**Review by youngnick — APPROVED** [review]
LGTM aside from one thing, so I'll leave this for another round of changes, sorry @robscott.

**Review by shaneutt — COMMENTED** [review]
/lgtm

File: mkdocs.yml
Comment (robscott) [reply mkdocs.yml:110]: Since this PR is big enough to conflict with ~everything docs related, I'm going to merge as is and make a follow up for this detail so I don't have to do any more rebases.

**Comment by robscott** [comment]
Will make a follow up PR to address @youngnick's final comment, but getting this in now to avoid more rebasing.

/hold cancel

**Pull request merged**
